File: javascript/contagem.js
/**
 * Recebe um array de números e conta quantos números inteiros existem
 * entre o primeiro e o último elemento do array de entrada.
 *
 * @param {number[]} arr
 * @returns {{ contagemIntervalo: number, contagemElementosArray: number }}
 */
function contarInteirosEntrePrimeiroEUltimo(arr) {
  if (!Array.isArray(arr) || arr.length === 0) {
    return { contagemIntervalo: 0, contagemElementosArray: 0 };
  }

  const primeiro = arr[0];
  const ultimo = arr[arr.length - 1];

  const inicio = Math.min(primeiro, ultimo);
  const fim = Math.max(primeiro, ultimo);

  // 1. Quantidade de inteiros no intervalo matemático estrito (inicio, fim)
  const minInt = Math.floor(inicio) + 1;
  const maxInt = Math.ceil(fim) - 1;
  const contagemIntervalo = Math.max(0, maxInt - minInt + 1);

  // 2. Quantidade de elementos inteiros presentes dentro do array no intervalo (inicio, fim)
  const contagemElementosArray = arr.filter(
    num => Number.isInteger(num) && num > inicio && num < fim
  ).length;

  return { contagemIntervalo, contagemElementosArray };
}

// Testes de exemplo
const testeArray1 = [2.5, 7, 3, 10, 4.2, 8]; // primeiro: 2.5, último: 8 -> inteiros no intervalo (2.5, 8): 3, 4, 5, 6, 7 (total: 5)
console.log("--- Contagem de Inteiros entre Primeiro e Último ---");
console.log("Array 1:", testeArray1);
console.log("Resultado 1:", contarInteirosEntrePrimeiroEUltimo(testeArray1));

const testeArray2 = [10, 15, 3.5, 2]; // primeiro: 10, último: 2 -> inteiros no intervalo (2, 10): 3, 4, 5, 6, 7, 8, 9 (total: 7)
console.log("Array 2:", testeArray2);
console.log("Resultado 2:", contarInteirosEntrePrimeiroEUltimo(testeArray2));

module.exports = { contarInteirosEntrePrimeiroEUltimo };

// Teste de sincronização com o GitHub3

File: java/Contagem.java
import java.util.Arrays;

public class Contagem {

    /**
     * Armazena as contagens de inteiros no intervalo numérico e elementos do array.
     */
    public static class Resultado {
        public final long contagemIntervalo;
        public final long contagemElementosArray;

        public Resultado(long contagemIntervalo, long contagemElementosArray) {
            this.contagemIntervalo = contagemIntervalo;
            this.contagemElementosArray = contagemElementosArray;
        }

        @Override
        public String toString() {
            return String.format("{ contagemIntervalo: %d, contagemElementosArray: %d }", 
                    contagemIntervalo, contagemElementosArray);
        }
    }

    /**
     * Recebe um array de números e conta quantos números inteiros existem
     * entre o primeiro e o último elemento do array de entrada.
     * 
     * @param arr Array de números
     * @return Objeto Resultado com a contagem do intervalo e dos elementos
     */
    public static Resultado contarInteirosEntrePrimeiroEUltimo(double[] arr) {
        if (arr == null || arr.length == 0) {
            return new Resultado(0, 0);
        }

        double primeiro = arr[0];
        double ultimo = arr[arr.length - 1];

        double inicio = Math.min(primeiro, ultimo);
        double fim = Math.max(primeiro, ultimo);

        // 1. Contagem no intervalo matemático estrito (inicio, fim)
        long minInt = (long) Math.floor(inicio) + 1;
        long maxInt = (long) Math.ceil(fim) - 1;
        long contagemIntervalo = Math.max(0, maxInt - minInt + 1);

        // 2. Contagem de elementos inteiros no array no intervalo (inicio, fim)
        long contagemElementosArray = 0;
        for (double num : arr) {
            if (num > inicio && num < fim && num == Math.floor(num)) {
                contagemElementosArray++;
            }
        }

        return new Resultado(contagemIntervalo, contagemElementosArray);
    }

    public static void main(String[] args) {
        double[] testeArray1 = {2.5, 7, 3, 10, 4.2, 8};
        System.out.println("--- Contagem de Inteiros entre Primeiro e Último ---");
        System.out.println("Array 1: " + Arrays.toString(testeArray1));
        System.out.println("Resultado 1: " + contarInteirosEntrePrimeiroEUltimo(testeArray1));

        double[] testeArray2 = {10, 15, 3.5, 2};
        System.out.println("Array 2: " + Arrays.toString(testeArray2));
        System.out.println("Resultado 2: " + contarInteirosEntrePrimeiroEUltimo(testeArray2));
    }
}
